### DISCUSSION TITLE
Windows best strategy to run pg_timetable

### DISCUSSION THREAD

**Discussion by RG40** [cybertec-postgresql/pg_timetable#397] [Q&A] [answered]
Hi,

I see 2 ways to run pg_timetable on windows server 2016.
1- Run pg_timetable.exe like a service, i didn't have success with this way, i created a .bat with the good command line, test the .bat, it's running well, created a service with path to the .bat. The service never started. I tried with the path directly to the .exe with parameters behind but it failed too. I miss something ?
2 - Autologin Administrator session when server start with logon scheduled task  running pg_timetable.exe, i don't like this way because i need to set an administrator windows password clearly in register...

Can you help me to define the best strategy for running pg_timetable when windows start ?

Thanks for your help

Rémy

**Comment by pashagolub** [comment, marked as answer]
Hello,

**pg_timetable** is a cross-platform application thus we have no specific OS functionality built in by default. 
That means one **cannot** run **pg_timetable** as a Windows service as is.

Of course, pull requests implementing Windows service support are welcome!

One will probably want to use [x/sys](https://pkg.go.dev/golang.org/x/sys/windows/svc) package or [go-svc](https://github.com/judwhite/go-svc) for that.

Another approach would be to use standalone service wrappers like [winsw](https://github.com/winsw/winsw).

Kind regards!
